back/engines/commercial/custom_maps/app/controllers/custom_maps/web_api/v1/map_configs_controller.rb

```diff
@@ -47,6 +47,29 @@ def show
           render json: serialized_map_config, status: :ok
         end
 
+        def duplicate_map_config_and_layers
+          authorize @map_config, :duplicate_map_config_and_layers?, policy_class: MapConfigPolicy
+
+          original_map_config = MapConfig.find(params[:id])
+          new_map_config = original_map_config.dup
+          new_map_config.mappable = nil
+
+          ActiveRecord::Base.transaction do
+            if new_map_config.save
+              new_map_config_layers = original_map_config.layers.map(&:dup)
+              new_map_config_layers.each do |layer|
+                layer.map_config = new_map_config
+                layer.save!
```

Comment:
My current thinking is to KISS and let the transaction fail hard here, if saving any layer raises an error. WDYT?

Reply:
Seems sensible to me. Realistically, there shouldn't be much that can fail here anyway